spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/DefaultSource.scala

```diff
@@ -0,0 +1,180 @@
+package org.apache.spark.sql.cassandra
+
+import org.apache.spark.{Logging, SparkConf}
+
+import org.apache.spark.sql.{DataFrame, SaveMode, SQLContext}
+import org.apache.spark.sql.SaveMode._
+import org.apache.spark.sql.sources.{CreatableRelationProvider, SchemaRelationProvider, BaseRelation, RelationProvider}
+import org.apache.spark.sql.types.StructType
+
+import com.datastax.spark.connector.cql.CassandraConnectorConf
+import com.datastax.spark.connector.rdd.ReadConf
+import com.datastax.spark.connector.writer.WriteConf
+
+import DefaultSource._
+
+/**
+ * Cassandra data source extends [[RelationProvider]], [[SchemaRelationProvider]] and [[CreatableRelationProvider]].
+ * It's used internally by Spark SQL to create Relation for a table which specifies the Cassandra data source
+ * e.g.
+ *
+ *      CREATE TEMPORARY TABLE tmpTable
+ *      USING org.apache.spark.sql.cassandra
+ *      OPTIONS (
+ *       c_table "table",
+ *       keyspace "keyspace",
+ *       cluster "test_cluster",
+ *       push_down "true",
+ *       spark_cassandra_input_page_row_size "10",
+ *       spark_cassandra_output_consistency_level "ONE",
+ *       spark_cassandra_connection_timeout_ms "1000"
+ *      )
+ */
+class DefaultSource extends RelationProvider with SchemaRelationProvider with CreatableRelationProvider with Logging {
```

Comment:
DefaultSource is the name conversion used by Spark Resources, so you don't need add class name to provider name when create a source. e.g. provider "org.apache.spark.sql.cassandra". If change it to CassandraSourceRelationProvider, provider "org.apache.spark.sql.casssandra.CassandraSourceRelationProvider". Spark SQL Internally Json, JDBC sources all use DefaultSource, so it's better to keep the  "DefaultSource"